app/models/server.rb

```diff
@@ -276,13 +280,27 @@ def self.find(id)
   end
 
   # Find the server with the lowest load (for creating a new meeting)
-  def self.find_available
+  def self.find_available(tag_arg = nil)
+    # Check if tag is required
+    tag = tag_arg
+    tag_required = false
+    if !tag.nil? && tag[-1] == '!'
+        tag = tag[0..-2] # always returns String, if tag is String
+        tag_required = true
+    end
+
+    # Find available&matching server with the lowest load
     with_connection do |redis|
+      ids = redis.zrange('server_load', 0, -1, with_scores: false)
+      if !tag.nil? && ids.none? { |myid| redis.hget(key(myid), 'tag') == tag }
+        raise RecordNotFound.new("Couldn't find available Server", name, nil) if tag_required
+        tag = nil # fall back to servers without tag
+      end
       id, load, hash = 5.times do
```

Comment:
Because I'm not sure for what scenario this 5-times loop has been added, I'm leaving it as is and repeat the server_load zrange query every time, to preserve the original behavior. I'd rather do it once above and instead of doing a 5-times loop here iterate through the zrange until a server with hash.present? has been found.